### ISSUE TITLE
Multiple upload support

### ISSUE THREAD

**Issue by flbulgarelli** [mumuki/mumuki-atheneum#712]
It would be nice that the solution upload support also one the following use cases: 

* Directory upload
* Multiple files upload
* A zip file upload

**Comment by faloi** [comment]
A quick solution for this could be a multiuploader that just concatenates all the files using some kind of separator. For instance:

```java
/**** A.java ****/
class A {
  private int field;

  public int getField() {
    return this.field;
  }
}

/**** B.java ****/
class B {
  public String shout(String something) {
    return something.toUpperCase();
  }
}
```

**Comment by flbulgarelli** [comment]
Yeah, I was actually thinking the same :smile:

**Comment by flbulgarelli** [comment]
Related to https://github.com/mumuki/mumuki-java-runner/issues/5

**Comment by faloi** [comment]
Ok. Any suggestions about the separators?

**Comment by flbulgarelli** [comment]
I'd try to use something that can not lead to ambiguities and that is very unlikely to be entered by a user. 

```java
/*@@@ B.java @@@*/
```

We could use `mumukit-directives` components to get the proper language-specific comment separator.

```ruby
"#{open_comment}@@@ #{filename}.#{extension} @@@#{close_comment}
```

**Comment by faloi** [comment]
Agree. Should this feature be configurable or automatically available for every exercise with upload?

**Comment by flbulgarelli** [comment]
I am not sure. I think we could enable this feature to every exercise, not only upload exercises. Or perhaps we could flag some exercises as _multi-editable_, to enable this behavior. See https://github.com/mumuki/mumuki-laboratory/issues/711.

This is a rough spec: 

1. If the exercise is of type `editor`, the student can click on a plus sign in the editor in order to create an additional editor. The student should be able to change the editor title.  When the student sends a solution with more than one editor, a single content solution is built in laboratory, separated as previously discussed. 
2. If the exercise is of type `upload`, and the uploaded file is a zip file, the file is unzipped and uploaded as a previously discussed. This could be done in the client.

**Comment by faloi** [comment]
Seems nice, but right now I only have time to develop part 2, the one related to the upload editor; and I really want to use this feature in about a month, on exercises with manual correction.

Can we proceed with that and let the multi-editor discussion for another time?

**Comment by flbulgarelli** [comment]
Sure!

**Comment by faloi** [comment]
Great. Do you think the zip file is necessary? Any advantages over a plain old `<input type="file" multiple>`?

**Comment by flbulgarelli** [comment]
I think that using a file is easier if you want to provide a small script to upload it automatically to the web, like course does. But your proposal works for me.
